DIDAddonsMLFeed.as

```diff
@@ -1,146 +1,196 @@
+            LaneProviderSettings settings;
+            settings.author = "MisfitMaid";
+            settings.internalName = "DID/CurrentCheckpoint";
+            settings.friendlyName = "Checkpoint counter";
+            return settings;
         }
 
-        class CurrenCheckpointHandler : LaneProvider {
-            LaneProviderSettings@ getProviderSetup() {
-                LaneProviderSettings settings;
-                settings.author = "MisfitMaid";
-                settings.internalName = "DID/CurrentCheckpoint";
-                settings.friendlyName = "Checkpoint counter";
-                return settings;
-            }
+        LaneConfig@ getLaneConfig(LaneConfig@ &in defaults) {
+            LaneConfig c = defaults;
+            if (plf is null) return c;
+            if (plf.spawnStatus == MLFeed::SpawnStatus::NotSpawned) return defaults;
+            c.content = Text::Format("%d", plf.CpCount % (mlf.CpCount+1))+" / "+Text::Format("%d", mlf.CpCount);
+            return c;
+        }
+    }
+
+    class NoRespawnTimeHandler : LaneProvider {
+        LaneProviderSettings@ getProviderSetup() {
+            LaneProviderSettings settings;
+            settings.author = "XertroV";
+            settings.internalName = "DID/NoRespawnTime";
+            settings.friendlyName = "No-Respawn Time";
+            return settings;
+        }
 
-            LaneConfig@ getLaneConfig(LaneConfig@ &in defaults) {
-                LaneConfig c = defaults;
-                const MLFeed::HookRaceStatsEventsBase_V3@ mlf = MLFeed::GetRaceData_V3();
-                const MLFeed::PlayerCpInfo_V3@ plf = mlf.GetPlayer_V3(MLFeed::LocalPlayersName);
-                if (plf.spawnStatus == MLFeed::SpawnStatus::NotSpawned) return defaults;
+        LaneConfig@ getLaneConfig(LaneConfig@ &in defaults) {
+            LaneConfig c = defaults;
+            if (plf is null) return c;
+            if (plf.spawnStatus == MLFeed::SpawnStatus::NotSpawned || plf.TimeLostToRespawns == 0) return defaults;
+            c.content = Time::Format(plf.TheoreticalRaceTime);
+            return c;
+        }
+    }
+
+    class NoRespawnSplitsHandler : LaneProvider {
+        LaneProviderSettings@ getProviderSetup() {
+            LaneProviderSettings settings;
+            settings.author = "XertroV";
+            settings.internalName = "DID/NoRespawnSplits";
+            settings.friendlyName = "No-Respawn Splits vs Personal Best";
+            return settings;
+        }
 
-                c.content = Text::Format("%d", plf.CpCount % (mlf.CpCount+1))+" / "+Text::Format("%d", mlf.CpCount);
-                return c;
-            }
+        LaneConfig@ getLaneConfig(LaneConfig@ &in defaults) {
+            LaneConfig c = defaults;
+            if (plf is null) return c;
+            if (plf.spawnStatus == MLFeed::SpawnStatus::NotSpawned || plf.TimeLostToRespawns == 0) return defaults;
+            auto cpMs = int(plf.BestRaceTimes[plf.cpCount-1]) - plf.LastTheoreticalCpTime;
```

Comment:
note: need to check best race times here to avoid index errors. both that cpCount > 0 and BestRaceTimes.Length >= cpCount.